File: frontend/src/app/model/request/UserStatusChangeRequest.ts
export default class UserStatusChangeRequest {
  constructor(
    /**
     * The new status of the user.
     */
    public status: number,
    /**
     * The username or full name of the user.
     */
    public username: string,
    /**
     * Url to the user's profile picture.
     */
    public imageUrl?: string,
  ) {}
}

File: spring-boot-backend/src/main/java/com/lucas/spring/model/request/user/UserStatusChangeRequest.java
package com.lucas.spring.model.request.user;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Stores information related to the status changes of the user.
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserStatusChangeRequest {
  /**
   * The new status of the user.
   */
  private Long status;
  /**
   * Url to the user's profile picture in url.
   */
  private String imageUrl;
  /**
   * The username or full name of the user.
   */
  private String username;
}
